contracts/TreasuryManager.sol

```diff
@@ -333,6 +338,7 @@ contract TreasuryManager is
                 false // Don't use internal balances
             )
         );
+        receivedBPT = BALANCER_POOL_TOKEN.balanceOf(address(sNOTE));
```

Comment:
This is not the received BPT, the received BPT is the difference before and after you join the pool.